packages/sdk-metrics/src/export/MetricReader.ts

```diff
@@ -139,9 +149,36 @@ export abstract class MetricReader {
       throw new Error('MetricReader is shutdown');
     }
 
-    return this._metricProducer.collect({
+    const collectionOptions = {
       timeoutMillis: options?.timeoutMillis,
-    });
+    };
+    const [sdkCollectionResults, ...additionalCollectionResults] =
+      await Promise.all([
+        this._sdkMetricProducer.collect(collectionOptions),
+        ...this._metricProducers.map(producer =>
+          producer.collect(collectionOptions)
+        ),
+      ]);
+
+    // Merge the results, keeping the SDK's Resource
```

Comment:
I think we should add multiple resource support to the metric reader and metric exporter instead of dropping resources silently -- the registered metric producer can be different meter providers that have different resources.

Reply:
I'm not sure how we could do that without making a breaking change. Also see [this part of the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.23.0/specification/metrics/sdk.md#registerproducermetricproducer:~:text=If%20the%20MeterProvider%20is%20an%20instance%20of%20MetricProducer%2C%20this%20MAY%20be%20used%20to%20register%20the%20MeterProvider%2C%20but%20MUST%20NOT%20allow%20multiple%20MeterProviders%20to%20be%20registered%20with%20the%20same%20MetricReader.):

> If the [MeterProvider](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.23.0/specification/metrics/sdk.md#meterprovider) is an instance of [MetricProducer](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.23.0/specification/metrics/sdk.md#metricproducer), this MAY be used to register the MeterProvider, but MUST NOT allow multiple [MeterProviders](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.23.0/specification/metrics/sdk.md#meterprovider) to be registered with the same MetricReader.

That's not 1:1 without implementation, but I think the spirit of it is to avoid this happening. I was actually thinking of making the MetricProducer only able to return ScopeMetrics to avoid the issue altogether.

Reply:
Fyi https://github.com/open-telemetry/opentelemetry-specification/pull/3636

Reply:
Is this OK with you for now @legendecas?